fecgov/fec-cms
# Feature/4104 Remove GTM Conditionals

## Summary

- Resolves #4104
Now that Google Tag Manager is in, let's remove the conditionals around it

## Impacted areas of the application

Visual changes: none
Data changes: none
Analytics changes: _extensive_

- removed reference to `FEC_FEATURE_USE_TAG_MANAGER` feature flag
  - global
- removed conditionals around `use_tag_manager` feature flag variable
  - global
- updated old GA code around custom events from `ga('notDAP.send', 'event', …`) to `analytics.customEvent(…)`
  - keyword-modal in legal search
  - by the numbers `.js-ga-event` clicks
  - data landing page `.js-ga-event` clicks
  - reaction box submitReaction events
  - analytics.pageView
    - election-search.js
    - urls.js/updateQuery()
    - urls.js/pushQuery()
  - vendor/tablist.js/show()
- updated widgets code from `ga('send'…)` to `analytics.customEvent(…)`
  - every pulldown change in aggregate totals
  - every interaction in contributions by state
  - every interaction in presidential finance map

## Screenshots

None

## Related PRs

None

## How to test

Could be tricky.

`dataLayer` is an array that GTM uses to queue events and share variables between the page and GTM. It has window-level scope ( `window.dataLayer = [ ]` )

- Pull, build, and run the branch like normal (pull, `npm i`, `npm run build`, `./manage.py runserver`).
- Edit middleware.py to add `"http://www.google-analytics.com/plugins/ua/linkid.js",` to the `script-src` section (Don't commit this. It's still only temporary so the GTM preview and debugger can work).
- Log in to Google Tag Manager.
  - In the workspace tab, the green box should list the latest version as at least 15.
  - Click Preview.
  - Another window/tab will open. This window is where you can track your DataLayer data & events
  - Use `http://127.0.0.1:8000/` plus whatever page you'd like. Unfortunately, we'll have to come back to here to test different pages.
  - Clicking Start will open _another_ window, which is where you'll interact with the site. You can use Inspector on this window to see console, network traffic, etc.
- Check your console (for the pop-up of the page to test) to make sure there are no errors that matter (images not loading don't matter).
- Clear the console if you'd like; it'll help while looking at different events.
- Switch to the Network tab to get it started–it may come in handy.
- Interact with the page in whatever way you're checking now.
- Make sure console is showing you the right events (GA should automatically switch to debug mode). Make sure the right eventCategory, eventLabel, eventAction, eventValue are being sent. GA should show you in console, but you can check network traffic, too (look for `collect`).
- If nothing else, looking at the most recent event's DataLayer tab back over in the Google Tag Assistant tab
- Page to check:
  - (coming soon)

**Notes**:
- This version of GTM Preview is super new!
- Pull-downs in the pop-up window may be out of place or not appear if you resize or move the pop-up. Moving the pop-up back to where it was, they should come back. It's weird.
- If needed, close the pop-up, switch back to the Google Tag Assistant tab, and click Reopen in the upper right corner
- Unfortunately, I think it's required that we close the pop-up and reopen it to a different page if we want to test another page. If not, tell me how!
____
